src/__test__/App.test.js

```diff
@@ -37,6 +40,25 @@ test('renders gadgets list', async () => {
   });
 });
 
+test('Empty Product List', async () => {
+  //Arrange
+
+  await setExpectations('src/__test__/'+EMPTY_GADGET_LIST_EXPECTATION_FILE);
+  process.env.REACT_APP_API_URL=STUB_URL
+
+  //Act
+  render(<App type="headphone" />);
+
+  //Assert
+  await waitFor(() => {
+    expect(screen.getByText("No Products found")).toBeInTheDocument();
+  })
+
+  emptyGadgetList.map(gadget => gadget.name).forEach((gadgetName) => {
```

Comment:
We can remove lines 57, 58, 59 because this loop will not run since the list will be empty